packages/go_router_builder/lib/src/route_config.dart

```diff
@@ -315,11 +315,15 @@ GoRouteData.\$route(
 
     final StringBuffer buffer = StringBuffer('queryParams: {\n');
 
-    for (final String param
-        in _ctorQueryParams.map((ParameterElement e) => e.name)) {
+    for (final ParameterElement param in _ctorQueryParams) {
+      final String parameterName = param.name;
+      String condition = '$parameterName != null';
+      if (param.hasDefaultValue) {
+        condition += '&& $parameterName != ${param.defaultValueCode!}';
```

Comment:
How do you tell the difference between a parameter with default value but explicitly set to null? in that case the URL it generate will always represents parameter with default value

Reply:
I was considering the type being non-nullable when a default value is set. But you are right, this is not always true. I assume we could allow / need `?param=null` in the URL

Reply:
yes, I can't think of other way to support nullable field without som caveats. Only allow non-nullable field to have default value SGTM

Reply:
So what do you think about the current implementation? Do you want me to change something?

Reply:
The more I think about this the more I think we probably shouldn't add this api. cc @johnpryan WDYT

Reply:
Maybe I can add some more insights:

I have to have a route that has a lot of parameters that are:
- No nullable
- Optional
- With a default value

For now, I have to do:

```dart
@TypedGoRoute<MyRoute>(path: '/my-route')
class MyRoute extends GoRouteData {
  const MyRoute({
    String? paramString,
    int? paramInt,
    // ... A lot more
  }): paramString = paramString ?? 'defaultValue',
    paramInt = paramInt ?? 0
    // ... A lot more
    ;

  final String paramString;
  final int paramInt;
  // ... A lot more
} 
```

The issue is that the URL becomes quite lengthy and not user-friendly.

If I do

```dart
context.go(const MyRoute().location);
```

The URL will be `/my-route?paramString=defaultValue&paramInt=0` and even longer with the other parameters.
I would like my URL to be `/my-route` only

Reply:
you can handle the default value in build method.

If we really want to do this, we need some kind of warning or even linting error if developer assign null to the nullable parameter.

Reply:
> If we really want to do this, we need some kind of warning or even linting error if developer assign null to the nullable parameter.

But didn't we say the parameter would be non-nullable? (https://github.com/flutter/packages/pull/2826#discussion_r1043635704)

Reply:
yes but what i meant was that we need something to warn developer about it.

Reply:
I can try to implement a warning or an error if the type is nullable

Reply:
What do you think about [🔊 Throw an error when a default value is used with a nullable type](https://github.com/flutter/packages/pull/2826/commits/3a9432f16c8f10cf673b1508060fc61902a28db7)?

Now it throws an error if a default value is used with a nullable type